tests/integration/helpers.py

```diff
@@ -372,3 +373,142 @@ def cluster_name(unit: Unit, model_name: str) -> str:
     output = json.loads(output.decode("utf-8"))
 
     return output[unit.name]["relation-info"][0]["application-data"]["cluster-name"]
+
+
+async def get_relation_data(
+    ops_test: OpsTest,
+    application_name: str,
+    relation_name: str,
+) -> list:
+    """Returns a that contains the relation-data.
+
+    Args:
+        ops_test: The ops test framework instance
+        application_name: The name of the application
+        relation_name: name of the relation to get connection data from
+    Returns:
+        a dictionary that contains the relation-data
+    """
+    # get available unit id for the desidered application
+    units_ids = [
+        app_unit.name.split("/")[1]
+        for app_unit in ops_test.model.applications[application_name].units
+    ]
+    assert len(units_ids) > 0
+    unit_name = f"{application_name}/{units_ids[0]}"
+    raw_data = (await ops_test.juju("show-unit", unit_name))[1]
+    if not raw_data:
+        raise ValueError(f"no unit info could be grabbed for {unit_name}")
+    data = yaml.safe_load(raw_data)
+    # Filter the data based on the relation name.
+    relation_data = [v for v in data[unit_name]["relation-info"] if v["endpoint"] == relation_name]
+    if len(relation_data) == 0:
+        raise ValueError(
+            f"no relation data could be grabbed on relation with endpoint {relation_name}"
+        )
+
+    return relation_data
+
+
+def get_read_only_endpoint(relation_data: list) -> Set[str]:
+    """Returns the read-only-endpoints from the relation data.
+
+    Args:
+        relation_data: The dictionary that contains the info
+    Returns:
+        a set that contains the read-only-endpoints
+    """
+    related_units = relation_data[0]["related-units"]
+    roe = set()
+    for _, r_data in related_units.items():
+        assert "data" in r_data
+        data = r_data["data"]["data"]
+
+        try:
+            j_data = json.loads(data)
+            if "read-only-endpoints" in j_data:
+                read_only_endpoints = j_data["read-only-endpoints"]
+                if read_only_endpoints.strip() == "":
+                    continue
+                for ep in read_only_endpoints.split(","):
+                    roe.add(ep)
+        except json.JSONDecodeError:
+            raise ValueError("Relation data are not valid JSON.")
+
+    return roe
```

Comment:
minor nit -> thoughts on renaming variables so they are named after what they represent? e.g. roe -> read_only_endpoints. ep -> endpoint. r_data -> relation_data, etc

I believe that this makes it easier to read the code for someone who is new to the codebase

Reply:
I renamed the variables.